src/test/java/com/google/cloud/flink/bigquery/BigQueryArrowFormatTest.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.table.catalog.CatalogTable;
+import org.apache.flink.table.catalog.CatalogTableImpl;
+import org.apache.flink.table.catalog.ObjectIdentifier;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.descriptors.DescriptorProperties;
+import org.apache.flink.table.factories.DynamicTableFactory.Context;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.RowKind;
+import org.junit.Test;
+
+public class BigQueryArrowFormatTest {
+
+  BigQueryArrowFormat bigQueryArrowFormat =
+      new BigQueryArrowFormat(Arrays.asList("word", "wordcount_output"));
+  MockScanContext mockScanContext = new MockScanContext();
+  static StreamTableEnvironment flinkTableEnv;
+  static String bigqueryReadTable;
+  static String srcQueryString;
+  static String flinkSrcTable;
+  static ObjectIdentifier tableIdentifier;
+  static TableSchema tableSchema;
+  public static final int DEFAULT_PARALLELISM = 10;
+  public static final String FLINK_VERSION = "1.11.0";
+  ImmutableMap<String, String> defaultOptions = ImmutableMap.of("table", "dataset.table");
+
+  public BigQueryArrowFormatTest() {
+    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+    flinkTableEnv = StreamTableEnvironment.create(env);
+
+    bigqueryReadTable = "q-gcp-6750-pso-gs-flink-22-01.wordcount_dataset.wordcount_output";
+    flinkSrcTable = "table1";
+    srcQueryString = "CREATE TABLE " + flinkSrcTable + " (word STRING , word_count BIGINT)";
+
+    tableIdentifier = ObjectIdentifier.of("bigquerycatalog", "default", flinkSrcTable);
+    tableSchema =
+        TableSchema.builder()
+            .field("word", DataTypes.STRING())
+            .field("word_count", DataTypes.BIGINT())
+            .build();
+  }
+
+  @Test
+  public void getChangelogModeTest() {
+    ChangelogMode result = bigQueryArrowFormat.getChangelogMode();
+    assertThat(result).isNotNull();
+    assertThat(result).isInstanceOf(ChangelogMode.class);
+    assertThat(result.getContainedKinds().toString()).isEqualTo("[INSERT]");
+    assertThat(result.contains(RowKind.INSERT)).isTrue();
+  }
+
+  @Test
+  public void createRuntimeDecoderTest() {
+    Context context = createContextObject();
+    final DataType producedDataType =
+        (context).getCatalogTable().getSchema().toPhysicalRowDataType();
+    DeserializationSchema<RowData> result =
+        bigQueryArrowFormat.createRuntimeDecoder(mockScanContext, producedDataType);
+    assertThat(result).isNotNull();
+    assertThat(result.getProducedType()).isNull();
+  }
+
+  public MockDynamicTableContext createContextObject() {
```

Comment:
Instead of creating this unwieldy mock object, why dont we use the Mockito library and mock the `Context` and `DataType` object?